federation/pkg/federation-controller/replicaset/replicasetcontroller_test.go

```diff
@@ -180,6 +175,81 @@ func TestReplicaSetController(t *testing.T) {
 	assert.Equal(t, rs.Status.AvailableReplicas, rs1.Status.AvailableReplicas+rs2.Status.AvailableReplicas)
 }
 
+func TestReplicaSetControllerPuppets(t *testing.T) {
```

Comment:
Will be great if you can add a comment here summarizing what this test does.
Creates a puppet federated RS and 2 RSs in underlying clusters and then verifies that the federated RS spec and status are updated as per the spec and status from RSs in underlying clusters.